### PR TITLE
Rewrite SSortNodesPredicate, *_wounded_predicate to modern C++11 lambda

### PR DESCRIPTION
**Pull request by GermanAizek**
@Xottab-DUTY

### CODE REVIEW

File: src/xrGame/agent_enemy_manager.cpp [hunk 1]
```diff
@@ -664,31 +634,24 @@ ALife::_OBJECT_ID CAgentEnemyManager::wounded_processor(const CEntityAlive* obje
...
-
-    IC bool operator()(const CAgentEnemyManager::WOUNDED_ENEMY& enemy) const { return (enemy.first == m_object); }
-};
-
 void CAgentEnemyManager::wounded_processor(const CEntityAlive* object, const ALife::_OBJECT_ID& wounded_processor_id)
 {
-    VERIFY(std::find_if(m_wounded.begin(), m_wounded.end(), find_wounded_predicate(object)) == m_wounded.end());
+    VERIFY(std::find_if(m_wounded.begin(), m_wounded.end(), [object](const WOUNDED_ENEMY& enemy)
+    {
+        VERIFY(object);
```
Comment (dreamer-dead) [inline src/xrGame/agent_enemy_manager.cpp]: Looks like this check will be executed every time the predicate is called, isn't it?
Original code did that check only one time in predicate constructor.

Comment (Xottab-DUTY) [inline src/xrGame/agent_enemy_manager.cpp]:
```suggestion
    VERIFY(object);
    VERIFY(std::find_if(m_wounded.begin(), m_wounded.end(), [object](const WOUNDED_ENEMY& enemy)
    {
```

File: src/xrGame/agent_enemy_manager.cpp [hunk 2]
```diff
@@ -664,31 +634,24 @@ ALife::_OBJECT_ID CAgentEnemyManager::wounded_processor(const CEntityAlive* obje
...
     m_wounded.emplace_back(object, std::make_pair(wounded_processor_id, false));
 }
 
 void CAgentEnemyManager::wounded_processed(const CEntityAlive* object, bool value)
 {
     VERIFY(value);
-    WOUNDED_ENEMIES::iterator I = std::find_if(m_wounded.begin(), m_wounded.end(), find_wounded_predicate(object));
+    WOUNDED_ENEMIES::iterator I = std::find_if(m_wounded.begin(), m_wounded.end(), [object](const WOUNDED_ENEMY& enemy)
+    {
+        VERIFY(object);
```
Comment (Xottab-DUTY) [inline src/xrGame/agent_enemy_manager.cpp]:
```suggestion
    VERIFY(object);
    WOUNDED_ENEMIES::iterator I = std::find_if(m_wounded.begin(), m_wounded.end(), [object](const WOUNDED_ENEMY& enemy)
    {
```

File: src/xrGame/agent_enemy_manager.cpp [hunk 3]
```diff
@@ -698,8 +661,11 @@ void CAgentEnemyManager::wounded_processed(const CEntityAlive* object, bool valu
 
 bool CAgentEnemyManager::wounded_processed(const CEntityAlive* object) const
 {
-    WOUNDED_ENEMIES::const_iterator I =
-        std::find_if(m_wounded.begin(), m_wounded.end(), find_wounded_predicate(object));
+    WOUNDED_ENEMIES::const_iterator I = std::find_if(m_wounded.begin(), m_wounded.end(), [object](const WOUNDED_ENEMY& enemy)
+    {
+        VERIFY(object);
```
Comment (Xottab-DUTY) [inline src/xrGame/agent_enemy_manager.cpp]:
```suggestion
    VERIFY(object);
    WOUNDED_ENEMIES::const_iterator I = std::find_if(m_wounded.begin(), m_wounded.end(), [object](const WOUNDED_ENEMY& enemy)
    {
```

**Review by Xottab-DUTY — APPROVED** [review]

**Pull request merged**